Realm.Shared/linq/RealmResultsVisitor.cs

```diff
@@ -363,6 +363,24 @@ private static void AddQueryEqual(QueryHandle queryHandle, string columnName, ob
                 NativeQuery.double_equal((QueryHandle)queryHandle, columnIndex, (double)value);
             else if (valueType == typeof(DateTimeOffset))
                 NativeQuery.datetime_seconds_equal(queryHandle, columnIndex, ((DateTimeOffset)value).ToUnixTimeSeconds());
+            else if (valueType == typeof(byte[]))
+            {
+                var buffer = (byte[])value;
+                if (buffer.Length == 0)
+                {
+                    // see RealmObject.SetByteArrayValue
+                    NativeQuery.binary_equal(queryHandle, columnIndex, (IntPtr)0x1, IntPtr.Zero);
```

Comment:
No, it just has to be anything but zero, as long as the length is zero. Seeing as how it's a dummy value I wouldn't bother with a constant.